homeassistant/components/squeezebox/switch.py

```diff
@@ -0,0 +1,192 @@
+                new_alarms = received_alarms - coordinator.known_alarms
+            removed_alarms = coordinator.known_alarms - received_alarms
+
+            if new_alarms:
+                for new_alarm in new_alarms:
+                    coordinator.known_alarms.add(new_alarm)
+                    _LOGGER.debug(
+                        "Setting up alarm entity for alarm %s on player %s",
+                        new_alarm,
+                        coordinator.player,
+                    )
+                    async_add_entities([SqueezeBoxAlarmEntity(coordinator, new_alarm)])
+
+            if removed_alarms and coordinator.available:
+                for removed_alarm in removed_alarms:
+                    _uid = f"{coordinator.player_uuid}-alarm-{removed_alarm}"
+                    _LOGGER.debug(
+                        "Alarm %s with unique_id %s needs to be deleted",
+                        removed_alarm,
+                        _uid,
+                    )
+
+                    entity_registry = er.async_get(hass)
+                    _entity_id = entity_registry.async_get_entity_id(
+                        Platform.SWITCH,
+                        DOMAIN,
+                        _uid,
+                    )
+                    if _entity_id:
+                        entity_registry.async_remove(_entity_id)
+                        coordinator.known_alarms.remove(removed_alarm)
+
+        _LOGGER.debug(
+            "Setting up alarm enabled entity for player %s", coordinator.player
+        )
+        # Add listener first for future coordinator refresh
+        coordinator.async_add_listener(_async_listener)
+
+        # If coordinator already has alarm data from the initial refresh,
+        # call the listener immediately to process existing alarms and create alarm entities.
+        if coordinator.data["alarms"]:
+            _LOGGER.debug(
+                "Coordinator has alarm data, calling _async_listener immediately for player %s",
+                coordinator.player,
+            )
+            _async_listener()
+        async_add_entities([SqueezeBoxAlarmsEnabledEntity(coordinator)])
+
+    entry.async_on_unload(
+        async_dispatcher_connect(hass, SIGNAL_PLAYER_DISCOVERED, _player_discovered)
+    )
+
+
+class SqueezeBoxAlarmEntity(SqueezeboxEntity, SwitchEntity):
+    """Representation of a Squeezebox alarm switch."""
+
+    _attr_translation_key = "alarm"
+    _attr_entity_category = EntityCategory.CONFIG
+
+    def __init__(
+        self, coordinator: SqueezeBoxPlayerUpdateCoordinator, alarm_id: str
+    ) -> None:
+        """Initialize the Squeezebox alarm switch."""
+        super().__init__(coordinator)
+        self._alarm_id = alarm_id
+        self._attr_available = True
+        self._attr_translation_placeholders = {"alarm_id": self._alarm_id}
+        self._attr_unique_id: str = (
+            f"{format_mac(self._player.player_id)}-alarm-{self._alarm_id}"
```

Comment:
just checked other platforms like button, we use `_` as delimiter, while here we use a normal dash, let's make it more consistent :)